ducc/src/object.rs

```diff
@@ -59,6 +60,62 @@ impl<'ducc> Object<'ducc> {
         }
     }
 
+    /// Defines a property using given key and descriptor
+    /// 
+    /// # Example
+    /// 
+    /// ```
+    /// let obj = ducc.create_object();
+    /// let get = ducc.create_function(|inv| Ok(24));
+    /// object.define_prop("prop", PropertyDescriptor::<()> {
+    ///     get: Some(get),
+    ///     ..Default::default()
+    /// }).unwrap();
+    /// ```
+    pub fn define_prop<K: ToValue<'ducc>, V: ToValue<'ducc>>(&self, key: K, desc: PropertyDescriptor<'ducc, V>) -> Result<()> {
+        let ducc = self.0.ducc;
+        let key = key.to_value(ducc)?;
+
+        let mut flags =
+            ffi::DUK_DEFPROP_HAVE_ENUMERABLE |
+            ffi::DUK_DEFPROP_HAVE_CONFIGURABLE;
+        if desc.writable {
+            flags |= ffi::DUK_DEFPROP_HAVE_WRITABLE | ffi::DUK_DEFPROP_WRITABLE;
```

Comment:
Duktape [will error](https://github.com/svaarala/duktape/blob/c91bcc9b1c05f65b2f6f7454faa43388ed92fb1d/src-input/duk_api_object.c#L547) if `HAVE_WRITABLE` is set when there is either a getter or a setter. It would be kinda nice to solve this on type level by either having `PropertyDescriptor` be a union type (for either getter/setter and value) or by having two functions with different parameters.

Reply:
Hmm yeah, like nesting a new type `PropertySource` inside `PropertyDescriptor` perhaps?

```rust
enum PropertySource<'ducc> {
    Value(Value<'ducc>),
    GetSet(Function<'ducc>, Function<'ducc>),
    Get(Function<'ducc>),
    Set(Function<'ducc>),
}
```

I think those variants make sense. I thought about `GetSet(Option<Function<'ducc>>, Option<Function<'ducc>>)` but it doesn't make sense to have `PropertySource::GetSet(None, None)`. Right? I'm not sure. Maybe you need more variants, like `PropertySource::None`?

And then you could have builder functions on the `PropertyDescriptor` type like `PropertyDescriptor::new().value("some value")` or `PropertyDescriptor::new().get_set(get_func, set_func)`.